### PR TITLE
feat: Update plugin with GetUIData interface

### PR DESCRIPTION
**Pull request by qiaoweiz**
resolve: #18

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,16 +3,16 @@ module github.com/abcxyz/jvs-plugin-jira
...
 require (
-	github.com/abcxyz/jvs v0.0.0-20230623161235-a02ba7d38f4b // TODO update to major version
+	github.com/abcxyz/jvs v0.0.0-20230718165522-9c429bb50dca // TODO update to major version
 	github.com/abcxyz/pkg v0.4.0
 	github.com/google/go-cmp v0.5.9
 	github.com/hashicorp/go-plugin v1.4.10
 )
 
 require (
-	cloud.google.com/go/compute v1.19.3 // indirect
```
Comment (dandans-verily) [inline go.mod:13]: Sanity check, are those auto-generated due to the `github.com/abcxyz/jvs` update?

File: pkg/plugin/config.go [hunk 2]
```diff
@@ -110,5 +114,21 @@ func (cfg *PluginConfig) ToFlags(set *cli.FlagSet) *cli.FlagSet {
 		Usage:   "The resource name of [google.cloud.secretmanager.v1.SecretVersion].",
 	})
 
+	f.StringVar(&cli.StringVar{
```
Comment (dandans-verily) [inline pkg/plugin/config.go:117]: I think we need to either give them default value or validate they are not empty. 

Here is an example of specifying default value: https://github.com/abcxyz/jvs/blob/main/pkg/config/justification_config.go#L106

File: pkg/plugin/plugin.go [hunk 3]
```diff
@@ -38,9 +35,16 @@ type issueMatcher interface {
 	MatchIssue(context.Context, string) (*validator.MatchResult, error)
 }
 
+// The JiraUIData comprises the data that will be displayed. At present, it exclusively includes the displayName and hint.
+type JiraUIData struct {
```
Comment (dandans-verily) [inline pkg/plugin/plugin.go:39]: Use jvspb.UIData directly instead of creating another struct?

Comment (sethvargo) [reply pkg/plugin/plugin.go:39]: +1 - or alias it `type JiraUIData = jvspb.UIData` if you need it to be local to the package.

File: pkg/plugin/plugin.go [hunk 4]
```diff
@@ -105,3 +120,22 @@ func secretVersion(ctx context.Context, secretVersionName string) (string, error
 
 	return string(resp.GetPayload().GetData()), nil
 }
+
+func GetUIDataFromPluginConfig(cfg *PluginConfig) JiraUIData {
```
Comment (yolocs) [inline pkg/plugin/plugin.go:124]: You can save this func if you set default values in the flags

File: go.mod (hunk 1 above)
Comment (qiaoweiz) [reply go.mod:13]: Yes.

File: pkg/plugin/config.go (hunk 2 above)
Comment (qiaoweiz) [reply pkg/plugin/config.go:117]: I updated with default value.

Comment (sethvargo) [reply pkg/plugin/config.go:117]: I'm not sure these make sense as default values. Default values should work "as-is" without customization. I don't think that's the default here.

Comment (dandans-verily) [reply pkg/plugin/config.go:117]: Agree that Hint default value "Jira Issue Key under specific project" is awkward here and we should require input and check not-empty. 

DisplayName default value "Jira Issue Key" seems reasonable? I don't have strong opinion on requiring a value

Comment (qiaoweiz) [reply pkg/plugin/config.go:117]: I am no sure if I understand correctly.
If the input is empty, then should we throw an error as: https://github.com/abcxyz/jvs-plugin-jira/blob/main/pkg/plugin/config.go#L57-L60? 

And @yolocs What do you think?

Comment (sethvargo) [reply pkg/plugin/config.go:117]: I think that's what @dandans-verily meant by:

> validate they are not empty.

Usually this is done in `Run` with something like if `c.flag == "" { return fmt.Errorf("missing required flag -flag") }`

Comment (yolocs) [reply pkg/plugin/config.go:117]: For hint, we might desire something more specific. Like "... under FOO project. Example: ..." And FOO would depend on your choice.

Comment (qiaoweiz) [reply pkg/plugin/config.go:121]: I keep the DisplayName with default value and update the Hint with not-empty check.

**Review by yolocs — APPROVED** [review]

**Review by yolocs — APPROVED** [review]

**Review by dandans-verily — APPROVED** [review]

**Pull request merged**